### PR TITLE
Add ability to override base job class for ActionMailer async delivery

### LINKED ISSUE #19793: Can't catch deserialization error when sending email with ActiveJob
For example:

```ruby
class UserMailer < ActionMailer::Base
  def payment_required(customer)
    # ...
  end
end
```

If I send delayed email:

```ruby
UserMailer.payment_required(customer).deliver_later(wait: 30.days)
```

And then user deletes his account the mailer fails with error `ActiveJob::DeserializationErrorError while trying to deserialize arguments: Couldn't find User with 'id'=2 `

So I can't cancel this job and I can catch this error to deal with it some way

### PR DESCRIPTION
**Pull request by latortuga**
Fixes #19793

### CODE REVIEW

File: actionmailer/test/message_delivery_test.rb [hunk 1]
```diff
@@ -93,4 +98,12 @@ def test_should_enqueue_and_run_correctly_in_activejob
       @mail.deliver_later(queue: :another_queue)
     end
   end
+
+  test "can override base job class" do
+    assert_performed_with(job: RescueJob, args: ['DelayedMailer', 'test_message', 'deliver_now', 1, 2, 3], queue: "another_queue") do
+      ActionMailer::Base.delivery_job_class = RescueJob
+      @mail.deliver_later(queue: :another_queue)
+      ActionMailer::Base.delivery_job_class = ActionMailer::DeliveryJob
```
Comment (senny) [inline actionmailer/test/message_delivery_test.rb]: Can we assign the original value back. Also we should do that in an ensure block.

```ruby
begin
  original_delivery_job_class = ActionMailer::Base.delivery_job_class
  ActionMailer::Base.delivery_job_class = RescueJob
  # ...
ensure
  ActionMailer::Base.delivery_job_class = original_delivery_job_class
end
```

**Comment by senny** [comment]
@latortuga can you also add an entry to the CHANGELOG?

**Comment by senny** [comment]
We recently introduced `config.action_mailer.deliver_later_queue_name` (#18587). This PR introduces a more powerful way to control Action Mailers integration into Active Job. I wonder wether we should still keep the queue name option around or if we should encourage subclassing `ActionMailer::DeliveryJob` for any kind of customizations.

/cc @rafaelfranca @cristianbica

File: actionmailer/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
+*   Allow override of base job class used for ActiveJob-powered mail delivery.
+
+    This allows you to catch errors in Job deserialization by using
+    `rescue_from` in your own base class.
+
+    `ActionMailer::Base.delivery_job_class = MyDeliveryJobClass`
+
```
Comment (senny) [inline actionmailer/CHANGELOG.md]: Would you mind adding:

> Fixes #19793.

Just above your name.

Comment (latortuga) [inline actionmailer/CHANGELOG.md]: I put in a blank line like your other changelog entries with similar annotations. Hope that's okay!

Comment (senny) [inline actionmailer/CHANGELOG.md]: great! :yellow_heart:

**Comment by latortuga** [comment]
Does this need more documentation, say in the guides somewhere or in comments?

**Comment by senny** [comment]
@latortuga good point about docs. You should extend the [configuring Rails guide](https://github.com/rails/rails/blob/master/guides/source/configuring.md) and the [ActionMailer::Base docs](actionmailer/lib/action_mailer/base.rb)

**Comment by latortuga** [comment]
Question for you, @senny. Based on the way I implemented it, will this automatically expose a `config.action_mailer.delivery_job_class` config variable?

**Comment by senny** [comment]
@latortuga not yet. Have a look at the ActionMailer::Railtie. Thats where the config variables are assigned.

**Comment by senny** [comment]
I just discussed this PR with the team. We see the situation as follows:

The configuration option `config.action_mailer.deliver_later_queue_name` addresses a common configuration case. If you happen to need more granular control over the job class we encourage you to reopen `ActionMailer::DeliveryJob` and attach your behavior.

@latortuga thanks you for the work on this PR :yellow_heart:

**Pull request closed without merging**